pkg/util/validation/exporter/exporter_test.go

```diff
@@ -136,77 +134,68 @@ func TestOverridesExporterWithRing(t *testing.T) {
 		return ring.NewDesc(), true, nil
 	}))
 
-	exporter, err := NewOverridesExporter(cfg, &validation.Limits{}, validation.NewMockTenantLimits(tenantLimits), log.NewNopLogger(), nil)
+	// Create an overrides-exporter.
+	cfg.Ring.InstanceID = "overrides-exporter-1"
+	cfg.Ring.InstanceAddr = "1.2.3.1"
+	e1, err := NewOverridesExporter(cfg, &validation.Limits{}, validation.NewMockTenantLimits(tenantLimits), log.NewNopLogger(), nil)
+	l1 := e1.ring.lifecycler
 	require.NoError(t, err)
-	require.NoError(t, services.StartManagerAndAwaitHealthy(ctx, exporter.subserviceManager))
-	t.Cleanup(func() { require.NoError(t, services.StopManagerAndAwaitStopped(ctx, exporter.subserviceManager)) })
+	require.NoError(t, services.StartAndAwaitRunning(ctx, e1))
+	t.Cleanup(func() { assert.NoError(t, services.StopAndAwaitTerminated(ctx, e1)) })
 
-	// Register this instance in the ring.
-	require.NoError(t, ringStore.CAS(ctx, ringKey, func(in interface{}) (out interface{}, retry bool, err error) {
-		desc := in.(*ring.Desc)
-		desc.AddIngester(
-			cfg.Ring.InstanceID,
-			fmt.Sprintf("%s:%d", cfg.Ring.InstanceAddr, cfg.Ring.InstancePort),
-			"",
-			[]uint32{1},
-			ring.ACTIVE,
-			time.Now(),
-		)
-		return desc, true, nil
-	}))
-
-	// Wait until the ring client observes the ring update.
-	var ringOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
-	test.Poll(t, time.Second, []string{exporter.ring.lifecycler.GetInstanceAddr()}, func() interface{} {
-		rs, _ := exporter.ring.client.GetAllHealthy(ringOp)
-		return rs.GetAddresses()
+	// Wait until it has received the ring update.
+	test.Poll(t, time.Second, true, func() interface{} {
+		rs, _ := e1.ring.client.GetAllHealthy(ringOp)
+		return rs.Includes(l1.GetInstanceAddr())
 	})
 
-	// This instance now owns the full ring, therefore overrides should be exported.
-	count := testutil.CollectAndCount(exporter, "cortex_limits_overrides")
-	assert.Equal(t, 10, count)
-
-	// Register a different instance.
+	// Set registeredTimestamp and (leader) token.
```

Comment:
I think you only want the guarantee the token 0 is owned by this replica (reason why you're changing it), but you shouldn't touch the `RegisteredTimestamp`. I would also comment here to explain why you're touching the token. This change is required to ensure that the leader will still be this replica once a 2nd replica will be started.

Reply:
A leftover as well, fixed.